tests/std/tests/P0466R5_layout_compatibility_and_pointer_interconvertibility_traits/test.cpp

```diff
@@ -0,0 +1,200 @@
+#pragma warning(push)
+#pragma warning(disable : 4408)
+        class D : public A {
+            union {};
+        };
+#pragma warning(pop)
+        class A1 : public A {};
+        class A2 : public A {};
+        class A3 : public A2 {};
+        class A4 : public A1, public A3 {};
+
+        ASSERT(is_pointer_interconvertible_base_of_v<A, A>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, B>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, const B>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, C>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, volatile C>);
+        ASSERT(is_pointer_interconvertible_base_of_v<volatile A, const C>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, D>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, A1>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, A2>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, A3>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A2, A3>);
+
+        ASSERT(!is_pointer_interconvertible_base_of_v<A1, A2>);
+        ASSERT(!is_pointer_interconvertible_base_of_v<A4, A>);
+        ASSERT(!is_pointer_interconvertible_base_of_v<B, C>);
+    }
+
+    // is_corresponding_member tests
+    {
+        struct S1 {
+            int v1;
+            int v2;
+        };
+
+        struct S2 {
+            int v1;
+            int v2;
+        };
+
+        struct S3 {
+            int v1;
+            int v2;
+            int v3;
+        };
+
+        struct S4 {
+            char v1;
+            int v2;
+            int v3;
+        };
+
+        struct S5 {
+            int v1;
+            int v2;
+            void* v3;
+        };
+
+        struct S6 {
+            int v1;
+            int v2;
+            double v3;
+        };
+
+        ASSERT(is_corresponding_member(&S1::v1, &S1::v1));
+        ASSERT(is_corresponding_member(&S1::v2, &S1::v2));
+        ASSERT(!is_corresponding_member(&S1::v1, &S1::v2));
+        ASSERT(!is_corresponding_member(&S1::v2, &S1::v1));
+        ASSERT(is_corresponding_member(&S1::v1, &S2::v1));
+        ASSERT(is_corresponding_member(&S1::v2, &S2::v2));
+        ASSERT(is_corresponding_member(&S1::v1, &S3::v1));
+        ASSERT(is_corresponding_member(&S1::v2, &S3::v2));
+        ASSERT(is_corresponding_member(&S5::v1, &S6::v1));
+        ASSERT(is_corresponding_member(&S5::v2, &S6::v2));
+
+        ASSERT(!is_corresponding_member(&S1::v1, &S2::v2));
+        ASSERT(!is_corresponding_member(&S1::v2, &S2::v1));
+        ASSERT(!is_corresponding_member(&S1::v1, &S4::v1));
+        ASSERT(!is_corresponding_member(&S1::v2, &S4::v2));
+        ASSERT(!is_corresponding_member(&S3::v1, &S4::v1));
+        ASSERT(!is_corresponding_member(&S3::v2, &S4::v2));
+        ASSERT(!is_corresponding_member(&S5::v1, &S6::v2));
+        ASSERT(!is_corresponding_member(&S5::v2, &S6::v1));
+        ASSERT(!is_corresponding_member(&S5::v3, &S6::v3));
+    }

```

Comment:
WG21-N4878 [meta.member]/4:

> "*Returns:* `true` if and only if `S1` and `S2` are standard-layout types, `M1` and `M2` are object types, `m1` and `m2` are not null, and `m1` and `m2` point to corresponding members of the common initial sequence (11.4) of `S1` and `S2`."

This indicates additional test cases:

* Non-standard-layout types
* Pointers to member functions
* Null pointers